test/test16.c

```diff
@@ -0,0 +1,43 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <unistd.h>
+#include "../Annotations/Annotations.h"
+
+struct my_struct {
+    char* x MustCall("free"); 
+    char* y MustCall("free"); 
+}; 
+
+void does_free(char* s CalledMethods("free")) {
```

Comment:
This function looks surprising to me, because the annotation is not what I'd expect. In the Java RLC, `CalledMethods("free")` on a variable means that that `free()` has definitely already been called on that variable's value. But here, you seem to be using it to mean "free() **will** be called on this variable", which is not quite the same.

There is a similar annotation in the Java version: `EnsuresCalledMethods`, which is added to a method, means that the annotated method guarantees that one of its arguments (or a field, etc.) has a particular `CalledMethods` fact true about it when the method finishes.